Add Elixir.load function

stephanvd opened the pull request:
Implements #262

- [x] Test this
- [x] Move start/load AST to seperate functions

stephanvd commented:
@bryanjos I'm not sure how to test this. How would you like to go about this?

bryanjos commented on lib/elixir_script/passes/create_js_modules.ex:
```diff
@@ -92,10 +92,30 @@ defmodule ElixirScript.Passes.CreateJSModules do
+        [],
+        JS.block_statement([
+          JS.return_statement(
+            JS.member_expression(
```
This should be 

```elixir
JS.call_expression(
  JS.member_expression(
    JS.identifier(:module),
    JS.identifier("__load")
  ),
  [JS.identifier("Elixir")]
)
```

Calling `__exports` directly won't work because it may not have been loaded yet

bryanjos commented:
@stephanvd good question. I think the easiest thing to test is the translation, testing that the JavaScript you expect to be generated, is. The tests in `/tests/translator` have examples of how to do so. `assert_translation` will check that the string given is in the generated JavaScript. I think that should be enough for this.

As for testing if it works when executed, I've usually tested things like this in an example project. I would like a test setup that would run the generated js in a js test suite, but not sure of the best way to do so.

stephanvd commented:
@bryanjos Added the specs and fixed your feedback.

In the last commit I've also added a simple node app to test the generated js with chai. Is this something you want?

bryanjos commented:
@stephanvd I was trying to figure out a good way to test the generated js in the standard library and I think this is a good way to do it! Thanks!

My only feedback is that what is in the package.json you created was moved to the package.json file in the root of the project. It has mocha and chai as well, but probably older versions that should be updated. If you have the time to change that, it would be helpful, otherwise I can just merge this and update it myself.

Either way, this is awesome!

stephanvd commented:
@bryanjos Thanks. Now using the root package.json

Only need to figure out why travis is failing. Do you see what's wrong? I can look into it somewhere this weekend.

bryanjos commented:
@stephanvd I see the problem. It's because I forgot to add a step to run `npm run build` in the travis ci script to build the bootstrap js file. I can merge this and add that to it.
